packages/rest/test/unit/router/routing-table.regexp.unit.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {anOpenApiSpec} from '@loopback/openapi-spec-builder';
+import {get, getControllerSpec, param} from '@loopback/openapi-v3';
+import {
+  ShotRequestOptions,
+  expect,
+  stubExpressContext,
+} from '@loopback/testlab';
+import {ControllerRoute, Request, RoutingTable} from '../../..';
+import {RegExpRouter} from '../../..';
+
+describe('RoutingTable with RegExpRouter', () => {
+  it('does not fail if some of the parameters are not decorated', () => {
+    class TestController {
+      @get('/greet')
+      greet(prefix: string, @param.query.string('message') message: string) {
+        return prefix + ': ' + message;
+      }
+    }
+    const spec = getControllerSpec(TestController);
+    const table = givenRoutingTable();
+    table.registerController(spec, TestController);
+    const paths = table.describeApiPaths();
+    const params = paths['/greet']['get'].parameters;
+    expect(params).to.have.property('length', 1);
+    expect(params[0]).to.have.properties({
+      name: 'message',
+      in: 'query',
+      schema: {type: 'string'},
+    });
+  });
+
+  it('finds simple "GET /hello" endpoint', () => {
+    const spec = anOpenApiSpec()
+      .withOperationReturningString('get', '/hello', 'greet')
+      .build();
+
+    class TestController {}
+
+    const table = givenRoutingTable();
+    table.registerController(spec, TestController);
```

Comment:
Have you considered using route-base approach to avoid the complexity of dealing with controllers?